### PR TITLE
Data Science with Python : ID3 Classification Algorithm #1822

### LINKED ISSUE #1822: Data Science with Python : ID3 Classification Algorithm
Welcome to 'DSWP' Team, good to see you here

This issue will helps readers in gaining all the guidance that one needs to know about ID3 Classification Algorithm. Tutorial to ID3 Classification Algorithm and how it's applied using sample code.

To get assigned to this issue, add your serial numbers mentioned in the spreadsheet of "Data Science with Python", the approach one would follow and choice you prefer (Documentation, Audio, Video). You can go with all three or any number of options you're interested to work on.

If you had referred any resources, add them up in "DS Resources". Similarly if you had used datasets, include them in "DS Datasets".

Domain : Machine Learning

Mentors Assigned : Kareem Negm / Gauri Bharadwaj

Points to Note :

The issues will be assigned on a first come first serve basis, 1 Issue == 1 PR.

"Issue Title" and "PR Title should be the same. Include issue number along with it.

Changes should be made inside the Datascience_With_Python/ directory & Datascience_With_Python branch.

Follow Contributing Guidelines & Code of Conduct before start Contributing.

This issue is only for 'GWOC' contributors of 'DSWP' domain.

All the best. Enjoy your open source journey ahead. 😎

### PR DESCRIPTION
**Pull request by surjyaneeh**
<hr>

## Description 📜

Fixes #1822

<hr>

## Type of change 📝

- [x] Documentation (Content Creation in the form of codes or tutorials)

<hr>

## Domain of Contribution 📊

- [x] Datascience with Python

<hr>

## Checklist ✅

- [x] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.
- [x] I have performed a self-review of my own code or work.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generates no new warnings.
- [x] I'm GWOC'21 contributor

<hr>

### CODE REVIEW

**Comment by surjyaneeh** [comment]
Please review @prathimacode-hub

**Comment by Kareem-negm** [comment]
Hello @surjyaneeh ,I don't see any contributions (Docs , Code) to review
please add all require files

**Comment by surjyaneeh** [comment]
> Hello @surjyaneeh ,I don't see any contributions (Docs , Code) to review
> please add all require files

But sir, It's clearly visible in dswp > ml > Algorithms

https://github.com/surjyaneeh/winter-of-contributing/tree/Datascience_With_Python/Datascience_With_Python/Machine%20Learning/Algorithms/ID3%20Classification%20Algorithm

**Comment by Kareem-negm** [comment]
it's showing now, your work seems good 
I appreciate your effort @surjyaneeh  
I will take a look and when I notice anything I will review it

**Review by Kareem-negm — COMMENTED** [review]
it will be better if you made these links as more professional, it showing that you are organized 
just make a markdown  like [ID3 Decision Tree Classifier from scratch in Python](https://towardsdatascience.com/id3-decision-tree-classifier-from-scratch-in-python-b38ef145fd90#:~:text=%20ID3%20Decision%20Tree%20Classifier%20from%20scratch%20in,wind%20blows%20from%20the%20South.%20If...%20More%20)

File: Datascience_With_Python/Machine Learning/Algorithms/ID3 Classification Algorithm/README.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+4. It is useful in data exploration . We can create new features or variables with better power to predict target variables.
+
+## Conclusion :
+
+ID3 Classification Algorithm is a popular and powerful machine learning algorithm. The main aim of this algorithm is to build a decision tree that helps to prectict the value of the target attribute. 
+But if the tree is too deep then the decision tree built using ID3 will be overfitting. To avoid overfitting, many more powerful methods are used (for example: Random Forest, gradient boosting etc.) . To build a good foundation of learning data science and machine learning ID3 algorithm is very much useful.
+
+## References :
+
+1. https://towardsdatascience.com/id3-decision-tree-classifier-from-scratch-in-python-b38ef145fd90#:~:text=%20ID3%20Decision%20Tree%20Classifier%20from%20scratch%20in,wind%20blows%20from%20the%20South.%20If...%20More%20
```
Comment (Kareem-negm) [inline Datascience_With_Python/Machine Learning/Algorithms/ID3 Classification Algorithm/README.md:51]: This is what I'm talking about

**Comment by surjyaneeh** [comment]
> it will be better if you made these links as more professional, it showing that you are organized
> just make a markdown like [ID3 Decision Tree Classifier from scratch in Python](https://towardsdatascience.com/id3-decision-tree-classifier-from-scratch-in-python-b38ef145fd90#:~:text=%20ID3%20Decision%20Tree%20Classifier%20from%20scratch%20in,wind%20blows%20from%20the%20South.%20If...%20More%20)

I have done the changes.

**Review by prathimacode-hub — CHANGES_REQUESTED** [review]
Changes to be done 👍 

In README.md, instead of getting directly into topic start from how this term is derived. 
In .ipynb, at the end provide a overall summary and final outcomes of the work you had done.

Kindly work on it. @surjyaneeh

**Comment by surjyaneeh** [comment]
> Changes to be done 👍
> 
> In README.md, instead of getting directly into topic start from how this term is derived.
> In .ipynb, at the end provide a overall summary and final outcomes of the work you had done.
> 
> Kindly work on it. @surjyaneeh

I have done the changes.

**Review by prathimacode-hub — APPROVED** [review]
PR approved. 👍@surjyaneeh

**Pull request merged**